installer/pkg/config-generator/generator.go

```diff
@@ -61,6 +63,78 @@ func New(cluster config.Cluster) ConfigGenerator {
 	}
 }
 
+// maoOperatorConfig contains configuration for mao managed stack
+// TODO(enxebre): move up to "github.com/coreos/tectonic-config
+type maoOperatorConfig struct {
+	metav1.TypeMeta `json:",inline"`
+	Provider        string         `json:"provider"`
+	AWS             *awsConfig     `json:"aws"`
+	Libvirt         *libvirtConfig `json:"libvirt"`
+}
+
+type libvirtConfig struct {
+	ClusterName string `json:"clusterName"`
+	URI         string `json:"uri"`
+	NetworkName string `json:"networkName"`
+	Replicas    int    `json:"replicas"`
+}
+
+type awsConfig struct {
+	ClusterName      string `json:"clusterName"`
+	ClusterID        string `json:"clusterID"`
+	Region           string `json:"region"`
+	AvailabilityZone string `json:"availabilityZone"`
+	Image            string `json:"image"`
+	Replicas         int    `json:"replicas"`
+}
+
+func (c *ConfigGenerator) maoConfig() *maoOperatorConfig {
+	cfg := maoOperatorConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "machineAPIOperatorConfig",
+		},
+	}
+
+	cfg.Provider = tectonicCloudProvider(c.Platform)
+
+	switch c.Platform {
+	case config.PlatformAWS:
+		// TODO: Use actual RHCOS channel.
+		ami, err := rhcos.AMI("tested", c.Region)
+		if err != nil {
+			log.Fatalf("Failed to lookup RHCOS AMI: %v", err)
+			return nil
+		}
+
+		if len(c.AWS.EC2AMIOverride) > 0 {
+			ami = c.AWS.EC2AMIOverride
+		}
+
+		cfg.AWS = &awsConfig{
+			ClusterName:      c.Name,
+			ClusterID:        c.ClusterID,
+			Region:           c.Region,
+			AvailabilityZone: "",
+			Image:            ami,
+			Replicas:         c.NodeCount(c.Worker.NodePools),
+		}
+
+	case config.PlatformLibvirt:
+		cfg.Libvirt = &libvirtConfig{
+			ClusterName: c.Name,
+			URI:         c.Libvirt.URI,
+			NetworkName: c.Libvirt.Network.Name,
+			Replicas:    c.NodeCount(c.Worker.NodePools),
+		}
+
+	default:
+		log.Warnf("Unknown provider for machine-api-operator: %v", cfg.Provider)
```

Comment:
This seems like it should be an error. If the provider is unknown, no workers will be created.

Reply:
It actually isn't possible to get here as `tectonicCloudProvider()` will panic on an unkown provider. I just twitch when I see switch statements without a default case... I can remove this or make it fatal though.

Reply:
I would panic here as well to make it clear.

Reply:
Fixed.